manager/scheduler/filter.go

```diff
@@ -246,13 +246,28 @@ func (f *PlatformFilter) Check(n *NodeInfo) bool {
 	// check if the platform for the node is supported
 	nodePlatform := n.Description.Platform
 	for _, p := range f.supportedPlatforms {
-		if (p.Architecture == "" || p.Architecture == nodePlatform.Architecture) && (p.OS == "" || p.OS == nodePlatform.OS) {
+		if f.platformComparison(*p, *nodePlatform) {
```

Comment:
I guess `false`. Some supported platforms were specified in the `TaskSpec`, but we don't know if the node can satisfy them.